### PR TITLE
Fix `int` to `unsigned` overflow conversion

### PR DESCRIPTION
**Pull request by Yaraslaut**
Fixes https://github.com/contour-terminal/contour/issues/964 and https://github.com/contour-terminal/contour/issues/1065

### CODE REVIEW

File: src/vtrasterizer/DecorationRenderer.cpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ auto DecorationRenderer::createTileData(Decorator decoration, atlas::TileLocatio
         case Decorator::Underline: {
             auto const thickness_half = max(1u, unsigned(ceil(underlineThickness() / 2.0)));
             auto const thickness = thickness_half * 2;
-            auto const y0 = max(0u, unsigned(underlinePosition()) - thickness_half);
+            auto const y0 = max(0, (underlinePosition() - static_cast<int>(thickness_half)));
```
Comment (whisperity) [inline src/vtrasterizer/DecorationRenderer.cpp:156]: I believe this is the part where we can safely decide to stop overusing `auto` for everything and explicitly mentioning the expected type here for `y0` in the declaration itself...

**Comment by whisperity** [comment]
#1065 is indeed fixed by this patch. Comment about the code style still stands.

**Comment by Utkarsh-khambra** [comment]
Fixes #964 too.

File: src/vtrasterizer/DecorationRenderer.cpp (hunk 1 above)
Comment (Yaraslaut) [reply src/vtrasterizer/DecorationRenderer.cpp:156]: > I believe this is the part where we can safely decide to stop overusing `auto` for everything and explicitly mentioning the expected type here for `y0` in the declaration itself...

We want to have non-negative value in y0, so unsigned is the minimal type that we can use for y0, while for ADL for max function we want to have int as argument, so something like `max<int>(a,b)`  can be used to get proper value

**Review by christianparpart — APPROVED** [review]
Many thanks (especially for the patience!)

**Pull request merged**